js/web/lib/wasm/jsep/webgpu/ops/reduce-shared.ts

```diff
@@ -188,7 +192,7 @@ export const createReduceSharedProgramInfo = (
   // One work group is responsible for only one element of output.
   return {
     name,
-    shaderCache,
+    shaderCache: { hint: `${cacheKey};${workgroupSize}`, inputDependencies: ['type'] },
```

Comment:
i think this is one place that JSEP is different from WebGPU EP.

in JSEP, WG size is not included in cache by default, but WebGPU EP it is.

Suggest to add a comments here for reference.

Reply:
Done